src/Component/ProductPage.jsx

```diff
@@ -0,0 +1,87 @@
+import { useParams, Link } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { useState, useEffect } from "react";
+import { addItem } from "../slices/cartSlice";
+import { PRODUCTS } from "../data/products";
+import "../css/ProductPage.css";
+
+export default function ProductPage() {
+  const { id } = useParams();
+  const product = PRODUCTS.find((p) => String(p.id) === String(id));
+  const dispatch = useDispatch();
+
+  if (!product)
+    return <div className="product-not-found">Product not found.</div>;
+
+  const imagesRaw = [
+    ...(Array.isArray(product.images) ? product.images : []),
+    ...(product.image ? [product.image] : []),
+  ].filter(Boolean);
+  const images = Array.from(new Set(imagesRaw));
+
+  const [activeImg, setActiveImg] = useState(images[0]);
+  useEffect(() => {
+    setActiveImg(images[0]);
+  }, [product.id]); 
+
+  const handleAdd = () => {
+    dispatch({
+      type: addItem.type,
```

Comment:
Use one convention for dispatching on Redux store, function or json with type and payload